### DISCUSSION TITLE
PHP SDK: "Invalid JSON data in request body." when sending e-invoice

### DISCUSSION THREAD

**Discussion by wastzzz** [fattureincloud/api#151] [Q&A] [answered]
Good evening, getting this error message since a few days. What's going on? I am using the latest SDK.

Exception when calling IssuedEInvoicesApi->sendEInvoice: [422] Client error: `POST https://api-v2.fattureincloud.it/c/542736/issued_documents/241515228/e_invoice/send` resulted in a `422 Unprocessable Entity` response: {"error":{"message":"Invalid JSON data in request body.","validation_result":null}}

Thank you
Massimo

**Comment by emajo** [comment]
Hi @wastzzz, due to a new json validation implementation, to send an empty body to the send einvoice method you have to build a request like this:
`$send_request = new SendEInvoiceRequest(['data' => new SendEInvoiceRequestData()]);`

Thanks,
Emanuele.

**Comment by wastzzz** [comment, marked as answer]
Hi,It started working again without changing anything. Am I missing something?Thanks.On 9 Nov 2022, at 08:33, Emanuele Saccomandi ***@***.***> wrote:﻿
Hi @wastzzz, due to a new json validation implementation, to send an empty body to the send einvoice method you have to build a request like this:
$send_request = new SendEInvoiceRequest(['data' => new SendEInvoiceRequestData()]);
Thanks,
Emanuele.

**Reply by emajo** [reply]
Hi @wastzzz, since we had a lot of customers reporting this issue we decided to allow values such as "{}" and "[]" in the request body, so you don't have to modify your code to make it work.

Thanks,
Emanuele.

**Comment by wastzzz** [comment]
Alright!Thanks.On 10 Nov 2022, at 08:54, Emanuele Saccomandi ***@***.***> wrote:﻿
Hi @wastzzz, since we had a lot of customers reporting this issue we decided to allow values such as "{}" and "[]" in the request body, so you don't have to modify your code to make it work.
Thanks,
Emanuele.
